bokeh/models/plots.py

```diff
@@ -371,22 +416,45 @@ def _check_snapped_toolbar_and_axis(self):
     The (default) data range of the vertical dimension of the plot.
     """)
 
-    x_mapper_type = Either(Auto, String, help="""
-    What kind of mapper to use to convert x-coordinates in data space
-    into x-coordinates in screen space.
+    @classmethod
+    def _scale(cls, scale):
```

Comment:
note: doesn't handle case where user sets "x_mapper_type=None", though I'm not sure that anybody does that anyways.